util/fetch/src/client.rs

```diff
@@ -199,16 +230,34 @@ impl Client {
 		})
 	}
 
-	fn background_thread(tx_start: TxStartup, rx_proto: mpsc::Receiver<ChanItem>) -> io::Result<thread::JoinHandle<()>> {
+	fn background_thread(tx_start: TxStartup, rx_proto: mpsc::Receiver<ChanItem>, https_proxy_url: Option<String>) -> io::Result<thread::JoinHandle<()>> {
 		thread::Builder::new().name("fetch".into()).spawn(move || {
 			let mut core = match reactor::Core::new() {
 				Ok(c) => c,
 				Err(e) => return tx_start.send(Err(e)).unwrap_or(())
 			};
 
+			let connector = {
+				let https_connector = hyper_rustls::HttpsConnector::new(4, &core.handle());
+
+				match https_proxy_url.unwrap_or_default().parse() {
```

Comment:
I don't like ```unwrap_or_default``` and rely on that an empty String will not be parsed as a SockAddr.

I would much prefer something like:
```rust

				match https_proxy_url.map(|urll| url.parse()) {
					Some(Ok(proxy_url)) => {
						debug!(target: "fetch", "https proxy is {:?}", proxy_url);
						ProxyConnector::from_proxy(
							https_connector,
							Proxy::new(Intercept::Https, proxy_url),
						).unwrap()
					}
					_ => {
						debug!(target: "fetch", "https proxy is not set");
						ProxyConnector::new(https_connector).unwrap()
					}
				}

```

Reply:
Got it! Thanks!